gen/SimDevice.yml

```diff
@@ -39,65 +42,214 @@ functions:
   HAL_SetSimValueBoolean:
 classes:
   SimValue:
-    shared_ptr: true
+    shared_ptr: false
     methods:
       SimValue:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       GetValue:
+        ignore: true
       SetValue:
+        ignore: true
   SimDouble:
-    shared_ptr: true
+    shared_ptr: false
     force_no_trampoline: true
     methods:
       SimDouble:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       Get:
       Set:
   SimEnum:
-    shared_ptr: true
+    shared_ptr: false
     force_no_trampoline: true
     methods:
       SimEnum:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       Get:
       Set:
   SimBoolean:
-    shared_ptr: true
+    shared_ptr: false
     force_no_trampoline: true
     methods:
       SimBoolean:
         overloads:
           "":
+            ignore: true
           HAL_SimValueHandle:
+            param_override:
+              val:
+                name: handle
       Get:
       Set:
   SimDevice:
-    shared_ptr: true
+    shared_ptr: false
+    force_type_casters:
+    - wpi::SmallVector
     methods:
       SimDevice:
         overloads:
           "":
+            ignore: true
           char*:
           char*, int:
           char*, int, int:
           HAL_SimDeviceHandle:
       CreateValue:
+        ignore: true
       CreateDouble:
       CreateEnum:
         overloads:
           char*, bool, std::initializer_list<const char *>, int32_t:
             ignore: true
           char*, bool, wpi::ArrayRef<const char *>, int32_t:
+            cpp_code: |
+              [](SimDevice * self, const char * name, bool readonly, const wpi::SmallVector<std::string, 8> &options, int32_t initialValue) {
+                wpi::SmallVector<const char *, 8> coptions;
+                coptions.reserve(options.size());
+                for (auto s: options) {
+                  coptions.push_back(s.c_str());
+                }
+                return self->CreateEnum(name, readonly, coptions, initialValue);
+              }
+
       CreateBoolean:
 
 inline_code: |
+
+  cls_SimValue
+    .def_property_readonly("value", [](const hal::SimValue &self) -> py::object {
```

Comment:
I feel `get()` would be a more idiomatic way to spell this, especially given it's read-only.